### DISCUSSION TITLE
$queryString doesn't use route parameters on initial dehydrate

### DISCUSSION THREAD

**Discussion by brandonchurchill** [livewire/livewire#4986] [7. Bugs (archived)] [answered]
I have the following route which has two parameters:
`Route::get('user/{id?}/{pageView?}', \App\Http\Livewire\User::class);`

The Livewire controller has been configured with $queryString specified:
`protected $queryString = ['id', 'pageView'];`

If the parameters being passed are:
id => 70
pageView => 'information'

On the initial page load, Livewire will set the URL to:
❌  http://example.com/user/70/information?id=70&pageView=information

The expected result, which is already the result for subsequent Livewire dehydrate requests on the page:
✅  http://example.com/user/70/information

I believe the following changes would correct this bug, however my device cannot pass the tests (even before the change), so I'm hoping someone else can test and make the changes. Thank you.

[src/Features/SupportBrowserHistory.php](https://github.com/livewire/livewire/blob/3ce704ac3314fc8713f710ee1c4e865846569c19/src/Features/SupportBrowserHistory.php#L47-L57)

<img width="832" alt="Screen Shot 2022-05-24 at 2 30 45 pm" src="https://user-images.githubusercontent.com/44955905/169950074-8a428839-f850-47e9-8d1b-fda497116a2c.png">

**Comment by joshhanley** [comment, marked as answer]
@brandonchurchill query string is everything from the `?` onwards. What you are after is URI or route segments, so you shouldn't be adding them to the `$queryString` property.

If you have properties with the same names as the route segments, then Livewire already automatically updates the URI segments for you.

Hope this helps!

**Reply by brandonchurchill** [reply]
Thanks for the reply @joshhanley, that's exactly what I needed.

I still find it odd that $queryString is treated differently in the initial dehydrate than it is in subsequent dehydrate requests, however you have solved my issue so I'm happy. Thanks again. 🙂

**Reply by joshhanley** [reply]
@brandonchurchill no worries! 🙂

Yeah the query string code isn't great, it will all be getting re-vamped in the next major version.
